src/legacy/core_plugins/embeddable_api/public/np_ready/public/lib/panel/panel_header/panel_actions/change_view_action.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { i18n } from '@kbn/i18n';
+import { IAction } from 'src/plugins/ui_actions/public';
+import { ContainerInput, IContainer } from '../../../containers';
+import { ViewMode, GetEmbeddableFactory, GetEmbeddableFactories } from '../../../types';
+import { NotificationsStart } from '../../../../../../../../../../core/public';
+import { KibanaReactOverlays } from '../../../../../../../../../../plugins/kibana_react/public';
+import { openChangeViewFlyout } from '../../../panel/panel_header/panel_actions/add_panel/open_change_view_flyout';
+
+export const CHANGE_VIEW_ACTION = 'changeView';
+
+interface ExpandedPanelInput extends ContainerInput {
+  expandedPanelId: string;
+}
+
+interface ActionContext {
+  embeddable: IContainer;
+}
+
+function hasExpandedPanelInput(
+  container: IContainer
+): container is IContainer<{}, ExpandedPanelInput> {
+  return (container as IContainer<{}, ExpandedPanelInput>).getInput().expandedPanelId !== undefined;
+}
+
+export class ChangeViewAction implements IAction<ActionContext> {
+  public readonly type = CHANGE_VIEW_ACTION;
+  public readonly id = CHANGE_VIEW_ACTION;
+  public order = 11;
+
+  constructor(
+    private readonly getFactory: GetEmbeddableFactory,
+    private readonly getAllFactories: GetEmbeddableFactories,
+    private readonly overlays: KibanaReactOverlays,
+    private readonly notifications: NotificationsStart,
+    private readonly SavedObjectFinder: React.ComponentType<any>
+  ) {}
+
+  public getDisplayName() {
+    return i18n.translate('embeddableApi.panel.removePanel.replaceView', {
+      defaultMessage: 'Replace visualization',
+    });
+  }
+
+  public getIconType() {
+    return 'kqlOperand';
+  }
+
+  public async isCompatible({ embeddable }: ActionContext) {
+    const isPanelExpanded =
+      embeddable.parent &&
+      hasExpandedPanelInput(embeddable.parent) &&
+      embeddable.parent.getInput().expandedPanelId === embeddable.id;
+
+    return Boolean(
```

Comment:
Once in dashboard plugin, you'll want to do a check in here to make sure the container type is a dashboard.  It'd be great if we could do it generally but the logic is specific to dashboard, and I don't think there is a good way to do it generally.

Reply:
Ok, done.